openapi3filter/req_resp_decoder.go

```diff
@@ -224,20 +224,65 @@ func decodeStyledParameter(param *openapi3.Parameter, input *RequestValidationIn
 
 func decodeValue(dec valueDecoder, param string, sm *openapi3.SerializationMethod, schema *openapi3.SchemaRef) (interface{}, error) {
 	var decodeFn func(param string, sm *openapi3.SerializationMethod, schema *openapi3.SchemaRef) (interface{}, error)
-	switch schema.Value.Type {
-	case "array":
-		decodeFn = func(param string, sm *openapi3.SerializationMethod, schema *openapi3.SchemaRef) (interface{}, error) {
-			return dec.DecodeArray(param, sm, schema)
+	if len(schema.Value.AnyOf) > 0 {
+		for _, sr := range schema.Value.AnyOf {
+			value, err := decodeValue(dec, param, sm, sr)
+			if value == nil && err == nil {
+				continue
+			}
+			return value, err
+		}
+		return nil, nil
```

Comment:
Shouldn't decoding continue & verify the other possible fields sibling to `anyOf`?

Reply:
Do you mean here will check all of anyof schema rules? even if matched one schema ?

Reply:
I mean given the legal schema:
```yaml
anyOf:
- format: ipv4
- format: ipv4
- pattern: '^[a-zA-Z_]+$'
minLength: 3
```
its `minLength: 3` won't be used for decoding. I can't think of a better example right now but maybe you get the idea?

Reply:
according to current code ,i think this function only decodes value according different types. it can't check this detailed schema. maybe it checks these format when invoke "VisitJSON" this function. i don't sure this . this is just my guess !